### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by dipshojol**

### CODE REVIEW

**Review by aitchslash — COMMENTED** [review]
IMO, the two ways to approach the many-to-many relationship in the Follow model are either 1) embed all this data in the User (the NoSQL way) or 2) imitate an RMDBS with a bare-bones junction table.

From Aecio's request to separate Profile from User, I conclude that option 1 is out.  For an example of a junction table you can check out the CommentLikes Model.

**Comment by aitchslash** [comment]
I'd also suggest using ObjectID's rather than strings.

**Comment by dipshojol** [comment]
> I'd also suggest using ObjectID's rather than strings.

can you check now.

File: server/models/follow.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const FollowSchema = new Schema({
+    userId: {
+        type: String,
+        ref: 'User',
+    },
```
Comment (aitchslash) [inline server/models/follow.js]: If this is User._id it's not a String, it's a Schema.Types.ObjectID.  If it's the username that might be useful, and I'm not totally opposed to including it, BUT it'll require more work with any update.  If you decide to include username you need to 1) change the field's name to username and 2) add the real User._id.

File: server/models/follow.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    followers: [{
+        type: String,
+        ref: 'User',
+    }],
+    following: [{
+        type: String,
+        ref: 'User',
+    }]
```
Comment (aitchslash) [inline server/models/follow.js]: You're still embedding arrays of data.  My earlier comment stands and my concerns remain.  I still advise that you take a look at the CommentLikes model.  Were I to implement this, you'd have a Following model (as a noun, not a verb) with a Follower and a Followee, both User._ids, both indexed.  That's it.  I might add a date, Following._id would be generated automatically by the db and you're good to go.  Well, we still have to generate good searches in the routes, but that's the next step.

**Comment by dipshojol** [comment]
> Going in the right direction. Be sure to index followers & following. They're not longer plural so I would change the names to follower & followee. You no longer need the userId field as all queries will be done through one of these two.

followee added

**Comment by dipshojol** [comment]
@aeciorc can you review it

File: server/models/follow.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const FollowSchema = new Schema({
+    userId: {
+        type: Schema.Types.ObjectId,
+        ref: 'User',
+    },
```
Comment (aeciorc) [inline server/models/follow.js]: You no longer need this

**Review by aeciorc — APPROVED** [review]
The model looks good now, just remove the lines I commented. I'noticed you added some unrelated changes for your other feature here, next time make sure you checkout a new branch for each issue you're working on

**Pull request merged**